pydataapi/pydataapi.py

```diff
@@ -347,16 +347,38 @@ def build(self) -> Dict[str, Any]:
         return self.dict(skip_defaults=True, by_alias=True)
 
 
+def find_arn_by_resource_name(
+    resource_name: str, boto3_client: Optional[boto3.session.Session.client]
+) -> str:
+    if not boto3_client:
+        boto3_client = boto3.client('rds')
+    return boto3_client.describe_db_clusters(DBClusterIdentifier=resource_name)[
+        'DBClusters'
+    ][0]['DBClusterArn']
+
+
 class DataAPI(AbstractContextManager):
     def __init__(
         self,
-        resource_arn: str,
+        *,
         secret_arn: str,
+        resource_arn: Optional[str] = None,
+        resource_name: Optional[str] = None,
         database: Optional[str] = None,
         transaction_id: Optional[str] = None,
         client: Optional[boto3.session.Session.client] = None,
         rollback_exception: Optional[Type[Exception]] = None,
+        rds_client: Optional[boto3.session.Session.client] = None,
     ) -> None:
+        if resource_name:
+            if resource_arn:
+                raise DataAPIError(
+                    f'resource_name should be set without resource_arn. resource_arn: {resource_arn},'
```

Comment:
If they are both set is it really a problem? If both are set maybe we can just use arn?

Reply:
I think implicit behavior is not good for users.
Do you think about it?
documents may be cover the problem.

Reply:
That is a good point. Ok we can force users to set only one or other if you would like.